OpenRA.Mods.Common/AI/HackyAI.cs

```diff
@@ -626,6 +668,97 @@ void AssignRolesToIdleUnits(Actor self)
 			}
 
 			FindAndDeployBackupMcv(self);
+
+			if (--minCaptureDelayTicks <= 0)
+			{
+				minCaptureDelayTicks = Info.MinimumCaptureDelay;
+				QueueCaptureOrders();
+			}
+		}
+
+		IEnumerable<Actor> GetVisibleActorsBelongingToPlayer(Player owner)
+		{
+			foreach (var actor in GetActorsThatCanBeOrderedByPlayer(owner))
+				if (actor.CanBeViewedByPlayer(owner))
+					yield return actor;
+		}
+
+		IEnumerable<Actor> GetActorsThatCanBeOrderedByPlayer(Player owner)
+		{
+			foreach (var actor in World.Actors)
+				if (actor.Owner == owner && !actor.IsDead && actor.IsInWorld)
+					yield return actor;
+		}
+
+		void QueueCaptureOrders()
+		{
+			if (Info.CapturingActorTypes.Count == 0 || Player.WinState != WinState.Undefined)
+				return;
+
+			var capturers = unitsHangingAroundTheBase.Where(a => a.IsIdle && Info.CapturingActorTypes.Contains(a.Info.Name)).ToArray();
+			if (capturers.Length == 0)
+				return;
+
+			if (combatantPlayers == null)
+				combatantPlayers = World.Players.Where(p => p.InternalName != "Everyone").ToArray();
+
+			var randPlayer = combatantPlayers.Random(Random);
+
+			var targetOptions = Info.CheckCaptureTargetsForVisibility
+				? GetVisibleActorsBelongingToPlayer(randPlayer)
+				: GetActorsThatCanBeOrderedByPlayer(randPlayer);
+
+			var capturableTargetOptions = targetOptions
+				.Select(a => new CapturerTarget<CapturableInfo>(a, "CaptureActor"))
+				.Where(target => target.Info != null && capturers.Any(capturer => target.Info.CanBeTargetedBy(capturer, target.Actor.Owner)))
+				.OrderByDescending(target => target.Actor.GetSellValue())
```

Comment:
A better option IMO was to capture the closest capturable actor or a mixture of both. This will make the AI blindly aiming for conyards with engineers, trying to march through the base from midgame onwards.